### PR TITLE
Create Authentication view

### PR DESCRIPTION
**Pull request by udayasri**
- Create LoginViewController with layout anchors
- Add StorageProvider enum to manage persistence of the exposure credentials

### CODE REVIEW

File: RefApp/ViewControllers/LoginViewController.swift [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    let logoImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "redbee")
+        imageView.contentMode = .scaleAspectFill
+        return imageView
+    }()
+    
+    let usernameTextField: RBMTextField = {
+        let textfield = RBMTextField(placeHolderText: NSLocalizedString("Username", comment: ""))
+        return textfield
```
Comment (FredrikSjoberg) [inline RefApp/ViewControllers/LoginViewController.swift]: optionally, we could add

textfield.textContentType = UITextContentType.username

File: RefApp/ViewControllers/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    }()
+    
+    let passwordTextField: RBMTextField = {
+        let textfield = RBMTextField(placeHolderText: NSLocalizedString("Password", comment: ""))
+        textfield.isSecureTextEntry = true
```
Comment (FredrikSjoberg) [inline RefApp/ViewControllers/LoginViewController.swift]: and

textfield.textContentType = UITextContentType.password

File: RefApp/ViewControllers/LoginViewController.swift [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            return
+        }
+        
+        Authenticate(environment: environment)
+            .login(username: username, password: password)
+            .request()
+            .validate()
+            .response{ [weak self] in
+                if let error = $0.error {
+                    self?.popupAlert(title: error.message , message: error.localizedDescription, actions: [okAction], preferedStyle: .alert)
```
Comment (FredrikSjoberg) [inline RefApp/ViewControllers/LoginViewController.swift]: ExposureError provides a lot more details (no exposed through localizedDescription) which could be helpful in this context.

In previous implementations we have used something along the lines of:

```Swift
let message = "\(error.code) " + error.message + "\n" + (error.info ?? "")
self?.popupAlert(title: error.domain , message: message, actions: [okAction], preferedStyle: .alert)
```

File: RefApp/ViewControllers/LoginViewController.swift (hunk 1 above)
Comment (udayasri) [reply RefApp/ViewControllers/LoginViewController.swift]: Yes, lets add that too 👍

File: RefApp/ViewControllers/LoginViewController.swift (hunk 2 above)
Comment (udayasri) [reply RefApp/ViewControllers/LoginViewController.swift]: 👍

File: RefApp/ViewControllers/LoginViewController.swift (hunk 3 above)
Comment (udayasri) [reply RefApp/ViewControllers/LoginViewController.swift]: nice, seems more informative. I will add that.

**Pull request merged**